processing/src/main/java/org/apache/druid/query/aggregation/post/ExpressionPostAggregator.java

```diff
@@ -179,6 +169,14 @@ public String getName()
     return name;
   }
 
+  @Override
+  public ValueType getType()
+  {
+    // this is wrong, replace with Expr output type based on the input types once it is available
+    // but treat as string for now
+    return ValueType.STRING;
```

Comment:
Why STRING? (Instead of COMPLEX or whatever.)

What bad things could potentially happen if the type returned by this method is wrong? (The javadoc should explain this, ideally.)

Reply:
Since this is a post-aggregator, probably not a lot will go wrong in most cases. We will potentially predict an incorrect query result signature as having strings instead of whatever the actual expression output type is. I guess it should technically probably return `null` for now since it is truly unknown, so it will appear in the signature that way. I'm going to consider making this change.

`COMPLEX` could be ok to use here in some circumstances, but would not be correct i think for example, if the row signature for a subquery is computed and then the column from the subquery post-agg is used as an input to another expression (since expression selectors do not currently handle complex inputs). There might be other cases too.

Reply:
I've changed this to be `null` for now, rather than incorrectly calling it string, so it will continue to be treated as unknown, but did go ahead and wire things up for a follow-up PR to be able to add output type inference to expressions during post-agg decoration.

Reply:
Also i did not explicitly change the output contract of `PostAggregator.getType` to `@Nullable`, because I don't think that long term it should be allowed to be that, but could change it to that for now while it is true, and remove in the future once the expression post-agg knows its output type. 

Alternatively, I could have added an explicit output type json property, similar to expression virtual column declaration, but I think we want to deprecate those once output type inference is in place, so decided not to do this at this time, but am open to including it if anyone thinks it is necessary.

Reply:
I think it'd be good to make it `@Nullable` for now, with a note that we'd like it to stop being nullable in the future when X/Y/Z things are done.

Reply:
Ah, I did end up adding it as `@Nullable`, I forgot to update this comment. It is missing the note on it maybe not being like that in the future, can add.